### DISCUSSION TITLE
TLS Issues (can't tell if misconfigured or bugged)

### DISCUSSION THREAD

**Discussion by Absolutia** [processone/ejabberd#3647] [Questions] [answered]
For the last four months, I have been working on trying to get an XMPP server running. Not just ejabberd, I have seen similar issues with Prosody and other servers. I have had the most success with ejabberd so far but like every other server **something** doesn't seem to work correctly. I am at my wits end, everything **seems** to be configured properly. I have followed some dozen tutorials, some as many as five times each, figuring there was something wrong with how I set it up but they all tread over the same ground and offer no troubleshooting advice-- that I managed to get from StackOverflow and even then it hasn't been very useful.

Attempting to connect over port 5223 causes Pidgin to error out with "Server closed the connection".

The output of `ejabberdctl live` reveals nothing of the sort:
`2021-07-21 15:00:30.363935+00:00 [info] Start accepting TCP connections at 192.168.1.3:5222 for ejabberd_c2s`
`2021-07-21 15:00:30.364264+00:00 [info] Start accepting TLS connections at 192.168.1.3:5223 for ejabberd_c2s`
and on attempted connection:
`2021-07-21 15:03:47.797480+00:00 [info] (<0.616.0>) Accepted connection 192.168.1.2:51724 -> 192.168.1.3:5223`, after which, Pidgin displays error, and server remains silent.

My guess here is that either the connection is silently failing server-side (either due to misconfigured server or a bug) or that Pidgin is bugged. Before I go about filing bug reports or trying to derive a patch, however, I'm asking about it here just in case it **really is** something painfully obvious that I somehow missed.

If anyone needs additional information, I can provide. Any help would be much appreciated, I'm getting desperate at this point.

**Comment by badlop** [comment, marked as answer]
I've installed ejabberd from git source code, it's a recent version but this probably is irrelevant to your problem. And maintained the default configuration, like this:

```yaml
listen:
  -
    port: 5223
    ip: "::"
    tls: true
    module: ejabberd_c2s
    max_stanza_size: 262144
    shaper: c2s_shaper
    access: c2s
    starttls_required: true
```

Started ejabberd:
```
2021-07-21 22:04:06.483590+02:00 [info] ejabberd 21.04.136 is started in the node ejabberd@galleta in 3.17s
2021-07-21 22:04:06.490177+02:00 [info] Start accepting TCP connections at [::]:5222 for ejabberd_c2s
2021-07-21 22:04:06.488030+02:00 [info] Start accepting TLS connections at [::]:5223 for ejabberd_c2s
```

And registered an account:
```
❯ ejabberdctl register user1 localhost somepass
User user1@localhost successfully registered
```

Then I've installed Pidgin 2.14.5 and configured an account:

- Protocol: XMPP
- Username: user1
- Domain: localhost
- Resource: pidd
- Password: somepass

In the Advanced tab:

- Security: Old SSL
- Connection port: 5223

When click on Add, it connects immediately:
```
2021-07-21 22:10:57.631925+02:00 [info] (<0.572.0>)
 Accepted connection [::ffff:127.0.0.1]:38302 -> [::ffff:127.0.0.1]:5223
2021-07-21 22:10:57.819732+02:00 [info] (tls|<0.572.0>)
 Accepted c2s SCRAM-SHA-1 authentication for user1@localhost by mnesia backend from ::ffff:127.0.0.1
2021-07-21 22:10:57.825617+02:00 [info] (tls|<0.572.0>)
 Opened c2s session for user1@localhost/pidd
```

**Reply by Absolutia** [reply]
It works.
I honestly can't believe that I didn't think to check the SSL option earlier.
Something that has me slightly confused though-- Pidgin is set to use SSL but the terminal output is showing `(tls|<0.575.0>)`.
So is it using SSL or TLS? Or is "TLS" just a catch-all term here?

**Comment by licaon-kter** [comment]
@Absolutia Until Pidgin 3 is out don't use old Pidgin 2 for testing as its features are known to be incomplete or even broken.

Use Dino or Gajim or Conversations or Siskin or Beagle.
